packages/react-app/src/components/pages/Bounties/index.tsx

```diff
@@ -28,45 +31,99 @@ const FilterResultPlaceholder = ({ message }: { message: string }): JSX.Element
 		justify="center"
 		align="center"
 	>
-		<Text	fontSize="lg">{ message }</Text>
+		<Text fontSize="lg">{message}</Text>
 	</Stack>
 );
 
+const baseFilters: FilterParams = {
+	search: '',
+	status: 'Open',
+	gte: 0,
+	lte: Infinity,
+	sortBy: 'reward',
+	asc: false,
+	customerId: BANKLESS.customerId,
+};
+
+const useDynamicUrl = (filters: FilterParams): string => {
+	const { customer } = useContext(CustomerContext);
+	const debounceSearch = useDebounce(filters.search, 500, true);
+
+	return useMemo(() => {
+		const { status, lte, gte, sortBy, asc: sortAscending } = filters;
+
+		let urlQuery = `?status=${status === '' ? 'All' : status}`;
+		urlQuery += `&search=${debounceSearch}`;
+		urlQuery += `&lte=${lte}`;
+		urlQuery += `&gte=${gte}`;
+		urlQuery += `&sortBy=${sortBy}`;
+		urlQuery += `&asc=${sortAscending}`;
+		urlQuery += `&customerId=${customer.customerId ?? BANKLESS.customerId}`;
+		urlQuery += `&customerKey=${customer.customerKey ?? BANKLESS.customerKey}`;
+
+		return urlQuery;
+
+	}, [filters, BANKLESS, customer, debounceSearch]);
+};
+
+const usePaginatedBounties = (bounties: BountyCollection[] | undefined, page: number) => {
+	// splits bounties according to the maximum page size
+	return useMemo(() => bounties
+		? bounties.slice(PAGE_SIZE * page, Math.min(bounties.length, PAGE_SIZE * (page + 1)))
+		: []
+	, [bounties, page, PAGE_SIZE]);
```

Comment:
why is `bounties` one of the dependencies for `useMemo`?

Reply:
Best practice with react is to use exhaustive dependencies with useMemo, useCallback and useEffect.

https://stackoverflow.com/questions/58866796/understanding-the-react-hooks-exhaustive-deps-lint-rule